pkg/apis/configuration/validation/validation.go

```diff
@@ -156,6 +159,41 @@ func validateUpstreams(upstreams []v1alpha1.Upstream, fieldPath *field.Path, isP
 	return allErrs, upstreamNames
 }
 
+var validVariableParams = map[string]bool{
+	"error":          true,
+	"timeout":        true,
+	"invalid_header": true,
+	"http_500":       true,
+	"http_502":       true,
+	"http_503":       true,
+	"http_504":       true,
+	"http_403":       true,
+	"http_404":       true,
+	"http_429":       true,
+	"non_idempotent": true,
+	"off":            true,
+}
+
+// validateNextUpstream checks the values given for passing queries to a upstream
+func validateNextUpstream(nextUpstream string, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	var occur int
+	params := strings.Fields(nextUpstream)
+	for _, para := range params {
+		if !validVariableParams[para] {
+			allErrs = append(allErrs, field.Invalid(fieldPath, para, "not a valid parameter"))
+		}
+		for _, p := range params {
```

Comment:
this part - lines 186-190 could be greatly simplified if you use a Set. For example, see https://github.com/nginxinc/kubernetes-ingress/blob/e6d5cf47761f31d1218c688314b02880ed3c7e3b/pkg/apis/configuration/validation/validation.go#L228

Reply:
but this is to check if theres are duplicates what would make it a bad/wrong input, making it a set would removed the duplicates that the code is checking for, no?

Reply:
you can use a Set to check for duplicates. When you process a parameter and check that is is valid, you add it to a Set. for the next parameter, after you check it is valid, you see if it is in the set ... if it is in the set, you have a duplicate...

Reply:
ahh okay, thanks for clearing that up 👍